lookout/style/format/postprocess.py

```diff
@@ -104,62 +156,28 @@ def filter_uast_breaking_preds(
-            safe_preds.append(i)
-            continue
-        pred_string = "".join(INDEX_CLS_TO_STR[j] for j in
-                              feature_extractor.labels_to_class_sequences[pred])
-        # we don't filter predictions that are fixing quote types
-        if CLS_QUOTES.intersection(vn_y.value) and CLS_QUOTES.intersection(pred_string):
-            safe_preds.append(i)
-            continue
-        content_before = files[vn_y.path].content.decode("utf-8", "replace")
-        parsed_before = _parse_code(vnode_parents[id(vn_y)], content_before, stub, parsing_cache,
-                                    feature_extractor.language, node_parents, log, vn_y.path)
-        if parsed_before is None:
-            continue
-        parent_before, start, end = parsed_before
-        # when the input node value is NOOP i.e. an empty string, the replacement is restricted
-        # to the first occurence
-        output_pred = "".join(n.value for n in vnodes[cur_i:cur_i+2]).replace(vn_y.value,
-                                                                              pred_string, 1)
-        diff_pred_offset = len(pred_string) - len(vn_y.value)
         try:
-            # to handle mixed indentations, we include the `VirtualNode` following the predicted
-            # one in the output predicted string, and start the rest of the sequence one
-            # `VirtualNode` further to avoid its repetitions
-            content_after = content_before[:vn_y.start.offset] \
-                + output_pred \
-                + content_before[vn_y.start.offset + len(vn_y.value)
-                                 + len(vnodes[cur_i + 1].value):]
-        # in case the prediction to check corresponds to the last label of a file
-        except IndexError:
-            content_after = content_before[:vn_y.start.offset] \
-                + output_pred
-        content_after = content_after[start:end + diff_pred_offset]
-        parent_after, errors_after = parse_uast(
-            stub, content_after, filename="", language=feature_extractor.language)
-        if not errors_after:
-            if check_uasts_are_equal(parent_before, parent_after):
-                safe_preds.append(i)
-    log.info("Non UAST breaking predictions: %d selected out of %d",
-             len(safe_preds), y_pred.shape[0])
+            vnodes_index = vnodes.index(vn_y, vnodes_index)
+        except ValueError:
+            raise AssertionError("vnodes_y and vnodes are not consistent.")
+        to_check.append((gt, pred, vn_y, i, vnodes_index))
+
+    parsing_cache = {}  # type: Dict[int, Optional[Tuple[bblfsh.Node, int, int]]]
+    with ThreadPool() as pool:
+        are_safe = pool.starmap(
+            partial(_check_vnode, vnodes=vnodes, files=files, feature_extractor=feature_extractor,
+                    stub=stub, vnode_parents=vnode_parents, node_parents=node_parents,
+                    parsing_cache=parsing_cache),
+            to_check)
+    print(are_safe)
```

Comment:
Maybe you used this for debugging and forgot to remove it, no ?

Reply:
Indeed, thanks!